core/common/src/main/java/alluxio/collections/UniqueFieldIndex.java

```diff
@@ -25,26 +25,34 @@
  * @param <T> type of objects in this {@link IndexedSet}
  */
 @ThreadSafe
-class UniqueFieldIndex<T> implements FieldIndex<T> {
+public class UniqueFieldIndex<T> implements FieldIndex<T> {
   private final IndexDefinition<T> mIndexDefinition;
   private final ConcurrentHashMap<Object, T> mIndexMap;
 
   /**
    * Constructs a new {@link UniqueFieldIndex} instance.
+   *
+   * @param indexDefinition definition of index
    */
   public UniqueFieldIndex(IndexDefinition<T> indexDefinition) {
     mIndexMap = new ConcurrentHashMap<>(8, 0.95f, 8);
     mIndexDefinition = indexDefinition;
   }
 
   @Override
-  public void add(T object) {
+  public boolean add(T object) {
```

Comment:
@gpang ,thank you for the comments. I changed the add function back. Instead of returning false,  it would just return without doing anything when meeting two duplicated objects.